### PR TITLE
Closes #393

### LINKED ISSUE #393: River survey data import setting dialog improvement
River survey data file has optional four indices, as described in manual:
https://iric-gui-user-manual.readthedocs.io/en/latest/06/01_riversurvey.html

And currently, when the four indices exists, the center point of each cross section is set to be the middle of low water way, i. e. the middle position between point with Index 2 and point with Index 3.

But for some applications, for river survey data with four indices, we want to set the senter point of cross section to be the lowest point in low water way.

We want to modify the dialog so that we can set the lowest point in low water way, to be the center point of cross-section.

### PR DESCRIPTION
**Pull request by kskinoue0612**
This closes #393.
This pull request also include code modification for refactoring, to clean the code of riversurvey DLL. The modifications for refactoring are committed with log header "Refactoring: ".

You can test with the attached file below.

[issue-393.zip](https://github.com/i-RIC/prepost-gui/files/2290456/issue-393.zip)

### CODE REVIEW

**Comment by kskinoue0612** [comment]
Hi Scott,

Thanks for your information.

No, I can not do that on my environment. On my PC, QList looks like below, and that's why I've been refactoring to use std::vector instead of QList or QVector.

screenshot

I'll appreciate your help to make it work on my environment.

Thanks,

Keisuke

**Comment by scharlton2** [comment]
Hi Keisuke,

I think all I did was install qt-vs-addin-1.2.3-opensource.exe from https://download.qt.io/archive/vsaddin/qt-vs-addin-1.2.3-opensource.exe.mirrorlist

Here's the checksums of the file I have:
```
    File: qt-vs-addin-1.2.3-opensource.exe
  CRC-32: d3d2620d
     MD5: dda5e45f3e70d24b51e034fab695417f
   SHA-1: 0e0640687f6809730cbf000c4deea8b12549bd20
 SHA-256: fedee98bdfc57aad6f666a2cc2a82f482ad12f0bdf249d8221c89a3566d00b17
 ```

I'm not positive it will work for you since you are using the express edition.  I'm confident it works with Visual Studio 2013 Community (which I know doesn't have a Japanese version).  But, since you know English so well, you might want to try the community version.  You can download it using the links
at the bottom of this page:
https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2013-community-vs

I would try installing the qt-vs-addin-1.2.3-opensource first though.

Scott

**Comment by scharlton2** [comment]
Here is what my about dialog looks like:

image

**Comment by scharlton2** [comment]
If you decide to go with the community version you'll also be able to have C++ and Fortran projects in the same solution.

Scott

File: libs/geodata/riversurvey/geodatariversurveycrosssectionwindowgraphicsview.cpp [hunk 1]
```diff
@@ -1159,15 +1159,15 @@ void GeoDataRiverSurveyCrosssectionWindowGraphicsView::inspectLimits(bool* minli
...
+		right = alist.size() - 1;
 		*maxlimit = false;
 	}
 	double min1, min2, max1, max2;
 	min1 = alist[left].position() - alist[selectedPoints.first().row()].position();
 	max1 = alist[right].position() - alist[selectedPoints.last().row()].position();
-	min2 = 0 - alist.last().position();
-	max2 = 0 - alist.first().position();
+	min2 = 0 - (*(alist.begin() + alist.size() - 1)).position();
+	max2 = 0 - (*(alist.begin())).position();
```
Comment (scharlton2) [inline libs/geodata/riversurvey/geodatariversurveycrosssectionwindowgraphicsview.cpp]: I think this is easier to read (for lines 1169 and 1170):
	min2 = 0 - alist.front().position();
	max2 = 0 - alist.back().position();

File: libs/gridcreatingcondition/riversurvey15d/gridcreatingconditionriversurvey15d.cpp [hunk 2]
```diff
@@ -1075,7 +1075,7 @@ void GridCreatingConditionRiverSurvey15D::appendCrossSectionToGrid(GeoDataRiverC
 {
 	Structured15DGridWithCrossSectionCrossSection* crosssection = new Structured15DGridWithCrossSectionCrossSection(name, grid);
 	GeoDataRiverCrosssection::AltitudeList& alist = cs.AltitudeInfo();
-	double offset = alist.first().position();
+	double offset = (*(alist.begin())).position();
```
Comment (scharlton2) [inline libs/gridcreatingcondition/riversurvey15d/gridcreatingconditionriversurvey15d.cpp]: Easier (line 1078):
    double offset = alist.front().position();

**Comment by kskinoue0612** [comment]
Hi Scott,

Thanks for your comment,

I'm very sorry, I have not noticed your comments until today.
Usually, when a comment is added to issues or pull requests, I receive a mail to gmail mail box. But not this time. Not sure why.

Anyway, I'll read your comments, and do some action for this.

Thanks,
--
Keisuke Inoue

**Comment by kskinoue0612** [comment]
Hi Scott,

I agree that your code is easier to read. I have not known front() and back().
I'll modify the code, and push the branch again.

Thanks,

Keisuke

**Comment by kskinoue0612** [comment]
Hi Scott,

Yes, As you know, I'm now using Express Edition.
But because of SRTM issue, maybe I'm going to switch to Community Edition, or buy Commercial Edition. When I've switched, I'll try Qt add-in.

Thanks,

Keisuke

**Pull request merged**